### ISSUE TITLE
Client version-checking of match between client and server to prevent corruption of data

### ISSUE THREAD

**Issue by RafaPolit** [huridocs/uwazi#3192]
An outdated client could still make an API call to a newer-versioned Uwazi and cause problems.

**Comment by fnocetti** [comment]
I think, in the most common case, this is achieved by versioning the API, so the client hits the version it needs on purpose. While this is not feasible for us, we could borrow some ideas and, for example, have the client send the desired version on a header/url param on the request, and have the API respond an error message if the version requested is not compatible with the one running in the server. Given this, we could deprecate versions by adapting the criteria to determine compatibility.

**Comment by RafaPolit** [comment]
Yes, we took this approach on a previous project, the trick is how to pass the version when the client is loaded so that it works as expected without the need to manually add another version property in a config file.  I don't think this is going to be a difficult thing to figure out.
